Sources/LanguageServerProtocol/SupportTypes/CompletionItem.swift

```diff
@@ -11,7 +11,23 @@
 //===----------------------------------------------------------------------===//
 
 /// A single completion result.
-public struct CompletionItem: Codable, Hashable {
+public struct CompletionItem: TextDocumentRequest, RequestType, ResponseType, Codable, Hashable {
```

Comment:
Could you add a comment e.g.
```swift
/// CompletionItem is a request for the `completionItem/resolve` method and a response type to the `textDocument/completion` request.
```
It’s dual behavior confused me for quite a while.